src/Controller/ModelController.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2018 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+namespace App\Controller;
+
+use BEdita\SDK\BEditaClientException;
+use Cake\Event\Event;
+use Cake\Http\Response;
+use Cake\Network\Exception\UnauthorizedException;
+use Cake\Utility\Hash;
+use Psr\Log\LogLevel;
+
+/**
+ * Model controller: list, add, edit, remove modeling related resources
+ *
+ * Object types, relations, properties and property_types
+ *
+ * @property \App\Controller\Component\PropertiesComponent $Properties
+ */
+class ModelController extends AppController
+{
+    /**
+     * Resource type currently used
+     *
+     * @var string
+     */
+    protected $resourceType = null;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function initialize() : void
+    {
+        parent::initialize();
+
+        $this->loadComponent('Properties');
+
+        if (!empty($this->request)) {
```

Comment:
Can `$this->request` be  empty? It should always be populated `\Cake\Controller\Controller::__construct()`.
Maybe it should be:
```php
$resourceType = $this->request->getParam('resource_type');
if (!empty($resourceType)) {
    // etc...
}

```

Reply:
`$this->request` cannot be empty at this point, but Scrutinizer complains about that, maybe thers's come way to convince him :smiley_cat: 
Will remove the check and add `object_types`as default `resource_type` param.